libs/partners/upstage/langchain_upstage/parsers.py

```diff
@@ -0,0 +1,342 @@
+    env_key: Optional[str] = None,
+    default: Optional[str] = None,
+) -> str:
+    """Get a value from a param or an environment variable."""
+    if param is not None:
+        return param
+    elif env_key and env_key in os.environ and os.environ[env_key]:
+        return os.environ[env_key]
+    elif default is not None:
+        return default
+    else:
+        raise ValueError(
+            f"Did not find {key}, please add an environment variable"
+            f" `{env_key}` which contains it, or pass"
+            f"  `{key}` as a named parameter."
+        )
+
+
+class LayoutAnalysisParser(BaseBlobParser):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        output_type: OutputType = "text",
+        split: SplitType = "none",
+    ):
+        """
+        Initializes an instance of the Upstage class.
+
+        Args:
+            output_type (OutputType, optional): The desired output type.
+                                                Defaults to "text".
+            split (SplitType, optional): The type of splitting to be applied.
+                                         Defaults to "none".
+            api_key (str, optional): The API key for accessing the Upstage API.
+                                     Defaults to None, in which case it will be
+                                     fetched from the environment variable
+                                     `UPSTAGE_API_KEY`.
+        """
+        self.api_key = get_from_param_or_env(
+            "UPSTAGE_API_KEY", api_key, "UPSTAGE_API_KEY"
+        )
+
+        self.output_type = output_type
+        self.split = split
+
+        validate_api_key(self.api_key)
+
+    def _get_response(self, files: Dict) -> Dict:
+        """
+        Sends a POST request to the API endpoint with the provided files and
+        returns the response.
+
+        Args:
+            files (dict): A dictionary containing the files to be sent in the request.
+
+        Returns:
+            dict: The JSON response from the API.
+
+        Raises:
+            ValueError: If there is an error in the API call.
+        """
+        try:
+            headers = {"Authorization": f"Bearer {self.api_key}"}
+            options = {"ocr": False}
+            response = requests.post(
+                LAYOUT_ANALYZER_URL, headers=headers, files=files, json=options
+            )
+            response.raise_for_status()  # HTTP 요청이 성공적으로 완료되었는지 확인
+
+            try:
+                result = response.json()  # JSON 디코딩
+            except json.JSONDecodeError as e:
+                raise ValueError(f"Failed to decode JSON response: {e}")
+        finally:
+            if "document" in files and not files["document"].closed:
+                files["document"].close()
+
+        return result
+
+    def _split_and_request(
+        self,
+        full_docs: fitzDocument,
+        start_page: int,
+        split_pages: int = LIMIT_OF_PAGE_REQUEST,
```

Comment:
```suggestion
        page_batch_size: int = DEFAULT_PAGE_BATCH_SIZE,
```

Split 이란 용어가 요청 보낼때 자르는 것과 결과를 자르는 것 두가지에 혼용되어 있어서 가독성이 꽤 떨어지는것 같습니다. 요청시 문서 페이지를 자르는것은 batch size 로 표현하면 어떨까요?